### DISCUSSION TITLE
PS 8.1.1 How to create a custom product type in PrestaShop?

### DISCUSSION THREAD

**Discussion by nukdokplex** [PrestaShop/PrestaShop#33875] [Q&A]
Hello, i'm developing module for my shop. So my question is how to create and add custom product type? There is 4 built-in types in prestashop, and i want to make my own 5th type.

It will be great if you provide some examples.

**Comment by MatShir** [comment]
That's a good idea ! For now, it seems we can't really add new type dynamically. Maybe some hack could be done. @jolelievre perhaps

**Comment by jolelievre** [comment]
Hi @nukdokplex 

indeed in the recent page we introduce this notion of product type because this information was lacking to optimize the rendering of the form and also to encore some domain rules that were missing. We didn't have time to look into how this notion could/should be extended though, mostly because we lacked actual use cases to imagine what should be doable.

Can you explain to us what is your use case exactly? And in which way do the current restrictions may block you?
